### PR TITLE
Adding CUDA-IPC UCT feature. WIP- do not review

### PR DESCRIPTION
**Pull request by Akshay-Venkatesh**
- adds put/get zcopy functionality
- adds gtest modifications to check cuda-ipc uct functionality
- caching logic in subsequent PR

@bureddy

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by miked-mellanox** [comment]
ok to test

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6869/ for details (Mellanox internal link).

File: test/gtest/uct/test_p2p_rma.cc [hunk 1]
```diff
@@ -69,7 +69,7 @@ ucs_status_t uct_p2p_rma_test::get_zcopy(uct_ep_h ep, const mapped_buffer &sendb
 void uct_p2p_rma_test::test_xfer(send_func_t send, size_t length,
                                  unsigned flags, uct_memory_type_t mem_type)
 {
-    mapped_buffer sendbuf(length, SEED1, sender(), 1);
+    mapped_buffer sendbuf(length, SEED1, sender(), 1, mem_type);
```
Comment (bureddy) [inline test/gtest/uct/test_p2p_rma.cc]: to make it compatible with other cuda TLs,  allocate send buffer of mem_type only for ipc.

```
+    uct_memory_type_t src_mem_type = UCT_MD_MEM_TYPE_HOST;
+
+
+    if ((GetParam()->tl_name.compare("cuda_ipc") == 0)) {
+        src_mem_type = mem_type;
+    }
+
+    mapped_buffer sendbuf(length, SEED1, sender(), 1, src_mem_type);

```

**Pull request closed without merging**